src/Domain/Details.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Domain;
+
+final class Details
+{
+    /** @var mixed */
+    private $original;
+
+    /** @var string */
+    private $file;
+
+    /** @var int */
+    private $line;
+
+    /** @var string  */
+    private $function;
+
+    /** @var string */
+    private $message;
+
+    public static function make(): Details
+    {
+        return new Details();
+    }
+
+    public function withFile(string $file): Details
```

Comment:
Something to keep in mind is:

- the `withSomething` approach is more used when you wanna reach immutability, and the returned value will be a new instance with the given file.
- In this case, you are setting a new value on the current instance, so you wanna rename this to `setFile`.

What do you think?

Reply:
Here is an example on my typescript code, where the withHeaders returns a new instance of the transporter with the new headers:

```ts
  public withHeaders(headers: { [key: string]: string }): TransporterContract {
    return new Transporter({
      headers,
      hosts: this.hosts,
      logger: this.logger,
      requester: this.requester,
      timeouts: this.timeouts,
    });
  }
```

Reply:
cc @olivernybroe @Jibbarth

Reply:
Yeah so I think using setter prob. makes sense here.
My original thought was actually a builder for the details class, but I didn't do that as it felt overkill.
@Jibbarth What do you think about this?

Reply:
That seem legit. 
I wasn't really aware about Immutable object in PHP, so after few blog posts reading, I'm agree with @nunomaduro that withBlabla is used for this case. 

The builder approach is also interresting, thanks @olivernybroe :+1: 

In this context, I think using setters is enough here.

Reply:
I have changed it to setters, thanks guys! :) 

If we want a builder instead we can always do that later on.